python/foglamp/services/core/api/plugins/update.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+import asyncio
+import os
+import logging
+import uuid
+import platform
+
+from aiohttp import web
+from foglamp.common import logger
+from foglamp.services.core import connect
+from foglamp.common.storage_client.payload_builder import PayloadBuilder
+from foglamp.services.core import server
+from foglamp.common.plugin_discovery import PluginDiscovery
+from foglamp.common.common import _FOGLAMP_ROOT, _FOGLAMP_DATA
+
+
+__author__ = "Ashish Jabble"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems Inc."
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_help = """
+    -------------------------------------------------------------------------------
+    | PUT             | /foglamp/plugin/{type}/{name}/update                      |
+    -------------------------------------------------------------------------------
+"""
+_logger = logger.setup(__name__, level=logging.INFO)
+
+
+async def update_plugin(request: web.Request) -> web.Response:
+    """ update plugin
+
+    :Example:
+        curl -X PUT http://localhost:8081/foglamp/plugins/south/sinusoid/update
+        curl -X PUT http://localhost:8081/foglamp/plugins/north/http_north/update
+    """
+    _type = request.match_info.get('type', None)
+    name = request.match_info.get('name', None)
+    try:
+        # TODO: filter and notification types will be handled in future
+        _type = _type.lower()
+        if _type not in ['north', 'south']:
+            raise ValueError("Invalid plugin type. Must be 'north' or 'south'")
+
+        # Check requested plugin name is installed or not
+        installed_plugins = PluginDiscovery.get_plugins_installed(_type, False)
+        installed_plugin_name = [p_name["name"] for p_name in installed_plugins]
+        if name not in installed_plugin_name:
+            raise KeyError("{} plugin is not yet installed. So update is not possible.".format(name))
+
```

Comment:
As per FOGL-3034, we should stop updating plugin only when category named `Installation` whose config item `upgradeOnInstall` value is false.
For above we don't want to hold this up, will be taken care itself in FOGL-3034